python/holidays/is-IS.py

```diff
@@ -0,0 +1,62 @@
+from dateutil.easter import EASTER_WESTERN
+
+from utils import SmartDayArrow
+from .holidays import Holiday, Locale
+
+
+class en_GB(Locale):
+    """
+    01-01: [NRF] Nýársdagur
+    3 days before Easter: [NRV] Skírdagur
+    2 days before Easter: [NRV] Föstudagurinn langi
+    Easter: [NRV] Páskadagur
+    1 day after Easter: [NRV] Annar dagur páska
+    39 days after Easter: [NRV] Uppstigningardagur
+    49 days after Easter: [NRV] Hvítasunnudagur
+    50 days after Easter: [NRV] Annar dagur hvítasunnu
+    05-01: [NF] Verkalýðsdagurinn
+    06-17: [NF] þjóðhátíðardagurinn
+    1. monday in August: [NV] Frídagur verslunarmanna
+    12-25: [NRF] Christmas Day
+    12-26: [NRF] Annar dagur jóla
+    """
+
+    locale = "is-IS"
+    easter_type = EASTER_WESTERN
+
+    def holiday_first_day_of_summer(self):
+        """
+        Calculate sumardagurinn fyrsti (first day of summer).
+
+        The holiday falls on the first Thursday after the 18th of April.
+        """
+        return [
+            Holiday(
+                locale=self.locale,
+                region="",
+                date=SmartDayArrow(self.year, 4, 18).shift_to_weekday("thursday"),
+                description="Sumardagurinn fyrsti",
+                flags="NV",
+                notes="",
+            )
+        ]
+
+    def holiday_half_days(self):
+        return [
+            Holiday(
+                locale=self.locale,
+                region="",
+                date=SmartDayArrow(self.year, 12, 24),
+                description="Aðfangadagur",
+                flags="NRF",
+                notes="Holiday from 13:00",
```

Comment:
The "Lög um frið vegna helgihalds" defines "Aðfangadagur" from 18:00, but I guess that is overruled by the definition in "Lög um 40 stunda vinnuviku" which defines it from 13:00...

Reply:
That's right. It's a church holiday from 18:00 but a public holiday from 13:00.

Reply:
Totally off-topic, but out of curiosity: does this gap matter somewhere or is it just an oddity?

Reply:
Another point: Both laws mention Christmas Eve as "Aðfangadagur jóla", however according to Wikpedia (https://is.wikipedia.org/wiki/A%C3%B0fangadagur) both names seem to be synonyms, and the shorter one more prevalent. What is your take on that?

Reply:
I think this is a formal/informal distinction. _Aðfangadagur_ would be, roughly, "the day before", but since it's only used for Christmas Eve everyone knows what you mean. I've changed it to the longer name in https://github.com/flother/holidata/commit/f08fea7c1546cdf72dd1e3f653dd600aba6e1e35.

Reply:
On the gap between 13:00 and 18:00, I think that is just an oddity. The church isn't my strongpoint so I couldn't say why, though.